paasta_tools/cli/cmds/spark_run.py

```diff
@@ -1028,6 +1031,18 @@ def paasta_spark_run(args):
         )
         return 1
 
+    # validate pool
+    if args.pool:
+        valid_pools = system_paasta_config.get_cluster_pools().get(args.cluster, [])
+        if valid_pools and args.pool not in valid_pools:
+            print(
+                PaastaColors.red(
+                    f"Invalid --pool value. List of valid pools: {valid_pools}"
+                ),
+                file=sys.stderr,
+            )
+            return 1
```

Comment:
suggestion: if this was a function, we could write a unit test for it

Reply:
Added function and unit tests